ghga_connector/core/exceptions.py

```diff
@@ -34,40 +34,53 @@ def __init__(self):
 class DirectoryDoesNotExistError(RuntimeError):
     """Thrown, when the specified directory does not exist."""
 
-    def __init__(self, *, output_dir: Path):
-        message = f"The directory {output_dir} does not exist."
+    def __init__(self, *, directory: Path):
+        message = f"The directory '{directory}' does not exist."
+        super().__init__(message)
+
+
+class PathIsNotDirectory(RuntimeError):
+    """Thrown when specified directory is not a directory"""
+
+    def __init__(self, *, directory: Path):
+        message = (
+            f"Path of output directory '{directory}' exists, but is not a directory."
```

Comment:
Since the message says "output directory", this class cannot be used for other paths, like the upload directory, so the name should be "OutputPathNotDirectory".

Reply:
Let's change it so it's agnostic, even if we're only using it for the output path right now,